src/main/kotlin/io/saagie/whyat/dao/AvroDao.kt

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright © 2017 Saagie (contact@saagie.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.saagie.whyat.dao
+
+import io.saagie.whyat.domain.Event
+import org.apache.avro.Schema
+import org.apache.avro.file.CodecFactory
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.FileSystem
+import org.apache.hadoop.fs.Path
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Repository
+import java.net.URI
+import java.time.format.DateTimeFormatter
+import javax.annotation.PostConstruct
+import org.apache.avro.generic.GenericRecord
+import org.apache.avro.file.DataFileWriter
+import org.apache.avro.generic.GenericDatumWriter
+import org.apache.avro.generic.GenericData
+import org.apache.avro.mapred.FsInput
+import org.apache.hadoop.fs.FSDataOutputStream
+import java.io.File
+
+@Repository
+class AvroDao : EventDao {
+
+    @Value("\${hdfs.url}")
+    var hdfsUrl = ""
+
+    @Value("\${hdfs.path}")
+    var hdfsPath = ""
+
+    var fs: FileSystem? = null
+
+    var configuration = Configuration()
+
+    var schema: Schema? = null
+
+    @PostConstruct
+    fun init() {
+        if (fs == null) {
+            val hdfsUri = "hdfs://$hdfsUrl"
+            configuration = configuration.apply {
+                // Set FileSystem URI
+                set("fs.defaultFS", hdfsUri)
+                // Because of Maven
+                set("fs.hdfs.impl", org.apache.hadoop.hdfs.DistributedFileSystem::class.java.name)
+                set("fs.file.impl", org.apache.hadoop.fs.LocalFileSystem::class.java.name)
+            }
+            // Set HADOOP user
+            System.setProperty("HADOOP_USER_NAME", "hdfs")
+            System.setProperty("hadoop.home.dir", "/")
+
+            fs = FileSystem.get(URI.create(hdfsUri), configuration)
+
+            createSchema()
+            schema = loadSchema()
+        }
+    }
+
+    @Synchronized
+    override fun storeEvent(event: Event) {
+        val dataFileWriter = prepareDataFileWriter()
+        try {
```

Comment:
You can do `prepareDataFileWriter().use(...)`, it will close the FileWriter for you because it implements Closeable